src/adapter/src/coord/ddl.rs

```diff
@@ -930,6 +968,27 @@ impl Coordinator {
         self.retire_compute_sinks(drop_sinks).await;
     }
 
+    /// Cleans pending cluster reconfiguraiotns for the identified connection
+    #[mz_ore::instrument(level = "debug")]
+    pub(crate) async fn retire_cluster_reconfigurations_for_conn(
+        &mut self,
+        conn_id: &ConnectionId,
+    ) {
+        let reconfiguring_clusters = self
+            .active_conns
+            .get(conn_id)
+            .expect("must exist for active session")
+            .pending_cluster_alters
+            .clone();
+        self.drop_reconfiguration_replicas(reconfiguring_clusters)
+            .await;
+        self.active_conns
+            .get_mut(conn_id)
+            .expect("must exist for active session")
+            .pending_cluster_alters
+            .clear();
```

Comment:
I'd guess there's a much cleaner way of doing this rust, I gave it a couple of failed iterations and moved on.

Reply:
Either 

```Rust
        let reconfiguring_clusters = self
            .active_conns
            .get_mut(conn_id)
            .expect("must exist for active session")
            .pending_cluster_alters
            .drain(..)
            .collect();
        self.drop_reconfiguration_replicas(reconfiguring_clusters)
            .await;
```

or

```Rust
        let reconfiguring_clusters = std::mem::take(self
            .active_conns
            .get_mut(conn_id)
            .expect("must exist for active session")
            .pending_cluster_alters);
        self.drop_reconfiguration_replicas(reconfiguring_clusters)
            .await;
```

would probably work.